### PR TITLE
Refactor short flag resolution

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- simplify short flag resolution using iterator-based candidate filtering
- add coverage for short flag selection and errors
- align compile-fail expectations with updated messages

## Testing
- `make fmt`
- `make lint`
- `make test`

------
https://chatgpt.com/codex/tasks/task_e_6893e471dce083228cea43878a58f8f7

## Summary by Sourcery

Refactor the short flag resolution to use iterator-based candidate filtering and simplify validation logic, update documentation example, and add comprehensive tests for short flag selection and error scenarios.

Enhancements:
- Simplify resolve_short_flag by iterating over default lowercase and uppercase candidates and filtering them by ASCII alphanumeric, reserved, and used sets
- Consolidate validation and error reporting for invalid, reserved, and duplicate CLI short flags into a single flow

Documentation:
- Update the doc comment example to use expect instead of unwrap for short flag resolution

Tests:
- Add rstest-based tests for default lowercase selection, uppercase fallback, and invalid, reserved, and duplicate cli_short error cases
- Align compile-fail UI test expectations for invalid and collision error messages

### CODE REVIEW

File: ortho_config_macros/src/derive/build.rs [hunk 1]
```diff
@@ -328,7 +330,68 @@ pub(crate) fn build_append_logic(fields: &[(Ident, &Type)]) -> proc_macro2::Toke
...
+            cli_short: Some('f'),
+            ..FieldAttrs::default()
+        };
+        let mut used = HashSet::from(['f']);
+        let err = resolve_short_flag(&name, &attrs, &mut used).expect_err("duplicate char");
+        assert!(
+            err.to_string()
+                .contains("short flag collision; supply `cli_short` to disambiguate")
+        );
+    }
```
[the remark this answers is not in the document]
Comment (leynos) [reply ortho_config_macros/src/derive/build.rs:394]: @coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix.

File: ortho_config_macros/src/derive/build.rs [hunk 2]
```diff
@@ -34,49 +34,54 @@ fn option_type_tokens(ty: &Type) -> proc_macro2::TokenStream {
...
+                format!("reserved `cli_short` '{user}' conflicts with global flags"),
+            ));
+        }
+        if !used_shorts.insert(user) {
+            return Err(syn::Error::new_spanned(
+                name,
+                "short flag collision; choose a different `cli_short` value",
+            ));
+        }
+        return Ok(user);
```
[the remark this answers is not in the document]
Comment (leynos) [reply ortho_config_macros/src/derive/build.rs:65]: @coderabbitai review

[the remark this answers is not in the document]
Comment (leynos) [reply ortho_config_macros/src/derive/build.rs:65]: @crafix
